src/x_net.c

```diff
@@ -918,16 +920,25 @@ static void netreceive_free(t_netreceive *x)
     netreceive_closeall(x);
 }
 
+void netreceive_class_cleanup(t_class *c)
+{
+    free(netreceive_buf);
+    netreceive_buf = 0;
+}
+
 static void netreceive_setup(void)
 {
     netreceive_class = class_new(gensym("netreceive"),
         (t_newmethod)netreceive_new, (t_method)netreceive_free,
         sizeof(t_netreceive), 0, A_GIMME, 0);
+    class_setfreefn(netreceive_class, netreceive_class_cleanup);
     class_addmethod(netreceive_class, (t_method)netreceive_listen,
         gensym("listen"), A_GIMME, 0);
     class_addmethod(netreceive_class, (t_method)netreceive_send,
         gensym("send"), A_GIMME, 0);
     class_addlist(netreceive_class, (t_method)netreceive_send);
+
+    netreceive_buf = malloc(INBUFSIZE);
```

Comment:
i have no clue about `PERTHREAD`, but is this guaranteed to allocate the buffer for each instance?

shouldn't there be a `(netreceive_buf==NULL)` check in `netsend_readbin()` instead

Reply:
you're totally right! d'oh... Actually, I have to do the same thing as in d_fft.c. Thanks for catching that!

Reply:
Just pushed a fix. Note that we still leak memory if `netsend_readbin()` is called from a different thread than `netreceive_setup()`. This can only happen with libpd, but not in "regular" Pd. The same problem also exists for `ooura_init` / `ooura_term`. 

In both cases, I'm asking myself whether it might be better to allocate the buffers directly somewhere in the Pd instance... e.g. `_instanceiter` for netreceive and `_instanceugen` for fft.